packages/react-next/src/components/Checkbox/Checkbox.base.tsx

```diff
@@ -1,152 +1,80 @@
-import { useId, useControllableValue, useMergedRefs } from '@uifabric/react-hooks';
-import { useFocusRects } from 'office-ui-fabric-react';
+import { useCheckbox } from './useCheckbox';
 
-const getClassNames = classNamesFunction<ICheckboxStyleProps, ICheckboxStyles>({
-  useStaticStyles: true,
-});
+const defaultSlots: Omit<ICheckboxSlots, 'root'> = {
+  input: 'input',
+  checkmark: Icon,
+  container: 'label',
+  text: 'span',
+  checkbox: 'div',
+};
 
-export const CheckboxBase = React.forwardRef((props: ICheckboxProps, forwardedRef: React.Ref<HTMLDivElement>) => {
-  const {
-    className,
-    disabled,
-    inputProps,
-    name,
-    boxSide = 'start',
-    theme,
-    ariaLabel,
-    ariaLabelledBy,
-    ariaDescribedBy,
-    styles,
-    checkmarkIconProps,
-    ariaPositionInSet,
-    ariaSetSize,
-    keytipProps,
-    title,
-    label,
-    onChange,
-  } = props;
+export const CheckboxBase = compose<'div', ICheckboxProps, {}, ICheckboxProps, {}>(
+  (props, forwardedRef, composeOptions) => {
+    const { slotProps, slots, state } = useCheckbox(props, composeOptions, forwardedRef);
+    const { disabled, keytipProps, label } = state;
 
-  const rootRef = React.useRef<HTMLDivElement | null>(null);
-  const mergedRootRefs = useMergedRefs(rootRef, forwardedRef);
-  const checkBox = React.useRef<HTMLInputElement>(null);
-  const [isChecked, setIsChecked] = useControllableValue(props.checked, props.defaultChecked, onChange);
-  const [isIndeterminate, setIsIndeterminate] = useControllableValue(props.indeterminate, props.defaultIndeterminate);
+    const onRenderLabel = (): JSX.Element | null => {
+      return label ? <slots.text {...slotProps.text}>{label}</slots.text> : null;
+    };
 
-  useFocusRects(rootRef);
-  useDebugWarning(props);
-  useComponentRef(props, isChecked, isIndeterminate, checkBox);
+    return (
+      <KeytipData keytipProps={keytipProps} disabled={disabled}>
+        {// tslint:disable-next-line:no-any
+        (keytipAttributes: any): JSX.Element => (
+          <slots.root {...slotProps.root}>
+            <slots.input
+              {...slotProps.input}
+              data-ktp-execute-target={keytipAttributes['data-ktp-execute-target']}
+              aria-describedby={mergeAriaAttributeValues(
+                slotProps.input['aria-describedby'],
+                keytipAttributes['aria-describedby'],
+              )}
+            />
+            <slots.container {...slotProps.container}>
+              <slots.checkbox {...slotProps.checkbox} data-ktp-target={keytipAttributes['data-ktp-target']}>
+                <slots.checkmark {...slotProps.checkmark} />
+              </slots.checkbox>
+              {(props.onRenderLabel || onRenderLabel)(props, onRenderLabel)}
+            </slots.container>
+          </slots.root>
+        )}
+      </KeytipData>
+    );
+  },
+  {
+    slots: defaultSlots,
+    displayName: 'CheckboxBase',
+    handledProps: [
+      'componentRef',
```

Comment:
I wish there were a more automatic way of doing this. After seeing this list. afraid that doing this in bulk will be more expensive than getNativeProps bundle size wise. There is no reusability here.